### PR TITLE
Implement EditorController:init

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/OpenLoco/src/Scenario.cpp [hunk 1]
```diff
@@ -499,7 +499,7 @@ namespace OpenLoco::Scenario
     static loco_global<ObjectManager::SelectedObjectsFlags*, 0x50D144> _inUseobjectSelection;
     static loco_global<ObjectManager::ObjectSelectionMeta, 0x0112C1C5> _objectSelectionMeta;
 
-    static std::span<ObjectManager::SelectedObjectsFlags> getInUseSelectedObjectFlags()
+    std::span<ObjectManager::SelectedObjectsFlags> getInUseSelectedObjectFlags()
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Scenario.cpp:502]: NB: a similar (if not identical) function exists in `Title.cpp`. Should this one be used there as well?

Comment (duncanspumpkin) [reply src/OpenLoco/src/Scenario.cpp:502]: I plan (not done yet) to make all users of these flags to own a unique_ptr of the flags. That is why there are multiple versions of this same function.

File: src/OpenLoco/src/EditorController.cpp [hunk 2]
```diff
@@ -25,14 +35,126 @@ using namespace OpenLoco::Ui;
...
+        if (wasTitleMode)
+            setScreenFlag(ScreenFlags::title);
+
+        setGameSpeed(GameSpeed::Normal);
+
+        auto& options = S5::getOptions();
+        auto& gameState = getGameState();
+
+        options.editorStep = Step::objectSelection;
+        options.difficulty = 0;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/EditorController.cpp:72]: 2

File: src/OpenLoco/src/EditorController.cpp [hunk 3]
```diff
@@ -25,14 +35,126 @@ using namespace OpenLoco::Ui;
...
+        gameState.lastLandOption = 0xFF;
+        gameState.lastMapWindowAttributes.flags = WindowFlags::none;
+
+        WindowManager::closeAllFloatingWindows();
+        initialiseViewports();
+        Title::sub_4284C8();
+        Audio::pauseSound();
+        Audio::unpauseSound();
+        ObjectManager::unloadAll();
+        ObjectManager::prepareSelectionList(false); // TODO: should probably be true, but crashes if it is?
```
[the remark this answers is not in the document]
Comment (AaronVanGeffen) [reply src/OpenLoco/src/EditorController.cpp:85]: Thanks, that explains that! I'll remove the comment.

File: src/OpenLoco/src/Scenario.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/Scenario.cpp:502]: Oh, good to know. Should I make another instance of the function as well then?

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Scenario.cpp]: I've gone ahead and done just that. We can revert 85fb9a1 if you prefer otherwise.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**